modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/appserver/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,290 @@
+        inputStreamHandler.start();
+        errorStreamHandler.start();
+        ClientConnectionUtil.waitForPort(Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) +
+                                         portOffset, DEFAULT_START_STOP_WAIT_MS, false,
+                                         automationContext.getInstance().getHosts().get("default"));
+        //wait until Mgt console url printed.
+        long time = System.currentTimeMillis() + 60 * 1000;
+        while (!inputStreamHandler.getOutput().contains(SERVER_STARTUP_MESSAGE) &&
+               System.currentTimeMillis() < time) {
+            // wait until server startup is completed
+        }
+        ClientConnectionUtil.waitForLogin(automationContext);
+        log.info("Server started successfully.");
+        return tempProcess;
+    }
+
+    /**
+     * This method is to shutdown a server
+     *
+     * @param portOffset        - port offset
+     * @param automationContext - AutomationContext
+     * @throws XPathExpressionException - Error when getting data from automation.xml
+     * @throws RemoteException          - Error when shutdown the server
+     */
+    public static void serverShutdown(int portOffset,
+                                      AutomationContext automationContext)
+            throws XPathExpressionException, RemoteException {
+        long time = System.currentTimeMillis() + DEFAULT_START_STOP_WAIT_MS;
+        log.info("Shutting down server..");
+        boolean logOutSuccess = false;
+        if (ClientConnectionUtil.isPortOpen(Integer.parseInt(ExtensionConstants.
+                                                                     SERVER_DEFAULT_HTTPS_PORT))) {
+
+            int httpsPort = Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset;
+            String url = automationContext.getContextUrls().getBackEndUrl();
+            String backendURL = url.replaceAll("(:\\d+)", ":" + httpsPort);
+
+            ServerAdminClient serverAdminServiceClient = new ServerAdminClient(backendURL,
+                              automationContext.getContextTenant().getTenantAdmin().getUserName(),
+                              automationContext.getContextTenant().getTenantAdmin().getPassword());
+
+            serverAdminServiceClient.shutdown();
+
+            while (System.currentTimeMillis() < time && !logOutSuccess) {
+                logOutSuccess = isServerDown(automationContext, portOffset);
+                // wait until server shutdown is completed
+            }
+            log.info("Server stopped successfully...");
+        }
+
+    }
+
+    /**
+     * This method is to merge two arrays together
+     *
+     * @param parameters - Server startup arguments
+     * @param cmdArray   - Server startup command
+     * @return - merged array
+     */
+    private static String[] mergePropertiesToCommandArray(String[] parameters, String[] cmdArray) {
+        if (parameters != null) {
```

Comment:
If the arrays are empty it will return an empty array i removed the null check condition